drivers/adc/adc_mcux_gau_adc.c

```diff
@@ -334,21 +389,85 @@ static int mcux_gau_adc_init(const struct device *dev)
 
 	ADC_Init(base, &adc_config);
 
-	if (ADC_DoAutoCalibration(base, config->cal_volt)) {
-		LOG_WRN("Calibration of ADC failed!");
-	}
-
 	ADC_ClearStatusFlags(base, kADC_DataReadyInterruptFlag);
 
 	config->irq_config_func(dev);
 	ADC_EnableInterrupts(base, kADC_DataReadyInterruptEnable);
+}
 
-	k_work_init(&data->read_samples_work, &mcux_gau_adc_read_samples);
+/* The driver saves the ADC configuration before entering a low-power state
+ * and restores it upon wake-up. This ensures that the ADC returns to its
+ * pre-sleep configuration, maintaining the exact state the application
+ * expects after resuming from a low-power mode.
+ *
+ * Specifically:
+ * - Hardware register states are saved before power-down
+ * - Calibration settings are preserved
+ * - Channel configurations are restored
+ * - Any ongoing conversions are properly handled
+ *
+ * NOTE: The driver adjusts the warmup time for the first read
+ * after wakeup to avoid getting garbage data. See mcux_gau_adc_do_read.
+ */
+static int mcux_gau_adc_pm_action(const struct device *dev,
+				  enum pm_device_action action)
+{
+	const struct mcux_gau_adc_config *config = dev->config;
+	struct mcux_gau_adc_data *data = dev->data;
+	ADC_Type *base = config->base;
+	int err = 0;
+
+	switch (action) {
+	case PM_DEVICE_ACTION_RESUME:
+		break;
+	case PM_DEVICE_ACTION_TURN_ON:
+		/* GAU is not enabled after PM3 wakeup. */
+		POWER_PowerOnGau();
+		mcux_gau_adc_init_common(dev);
+		if (data->restore_after_sleep) {
+			err = mcux_gau_adc_calibrate(dev, false);
+			base->ADC_REG_ANA = data->saved_state.adc_reg_ana;
+			base->ADC_REG_INTERVAL = data->saved_state.adc_reg_interval;
+			data->restore_after_sleep = false;
+		}
+		break;
+	case PM_DEVICE_ACTION_SUSPEND:
+		break;
```

Comment:
If we are enabling device PM, then since ADC is one of the biggest power consumers of the whole chip, I think we should try to take power saving actions on suspend, maybe by disabling the module and/or gating it's clock.